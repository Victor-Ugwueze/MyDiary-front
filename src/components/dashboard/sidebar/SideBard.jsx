// react libraries
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const SideBar = ({
  isSidebarOpen, onClick, currentView, openNewEntryModal
}) => (
  <div id='left-section'>
    <ul className={`nav-tabs ${isSidebarOpen ? 'show-sidebar' : 'hide-sidebar'} `} role='tablist'>
      <li>
        <div className='nav-item-row'>
          <div className='icon'>
            <img
              className='add-entry'
              data-target='add-new-entry'
              src='/images/icons/diary_add_white.png'
            />
          </div>
          <div className='text'>
            <h3 className='add-entry' data-target='add-new-entry' onClick={openNewEntryModal}>
              New Entry
            </h3>
          </div>
        </div>
      </li>
      <li>
        <div
          className={`nav-item-row nav-it ${currentView.diary && 'active'} dairy`}
          data-target='diary'
          onClick={onClick}>
          <img className='icon' src='/images/icons/diary_icon_list_white.png' data-target='diary' />
          <h3 className='text' role='tab' data-target='diary'>
            My Diary
          </h3>
        </div>
      </li>
      <li>
        <div
          className={`nav-item-row ${currentView.settings && 'active'} nav-it settings`}
          data-target='settings'
          onClick={onClick}>
          <img className='icon' src='/images/icons/settings_icon.png' data-target='settings' />
          <h3 className='text' role='tab' data-target='settings'>
            Settings
          </h3>
        </div>
      </li>
      <li>
        <div
          className={`nav-item-row nav-it get-profile ${currentView.profile && 'active'}`}
          data-target='profile'
          onClick={onClick}>
          <img className='icon' src='/images/icons/profile_icon_white.png' data-target='profile' />
            <h3 className='text' role='tab' data-target='profile'>
              Profile
            </h3>
        </div>
      </li>
      <div id='logout' className='nav-item-row nav-it'>
        <div className='icon'>
          <img src='/images/icons/logout.png' />
        </div>
        <div>
          <Link to='/logout'>Log out</Link>
        </div>
      </div>
    </ul>
  </div>
);

SideBar.propTypes = {
  isSidebarOpen: PropTypes.bool,
  onClick: PropTypes.func,
  currentView: PropTypes.object,
  openNewEntryModal: PropTypes.func,
};

export default SideBar;
