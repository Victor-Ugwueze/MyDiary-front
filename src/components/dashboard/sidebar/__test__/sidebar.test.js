// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


// components
import SideBard from '../SideBard';


configure({
  adapter: new Adapter(),
});

describe('Renders the sidebar component', () => {
  it('should render sidebar of dashboard page', () => {
    const component = shallow(
      <SideBard currentView={{ diary: true }}/>
    );
    expect(component).toMatchSnapshot();
  });
});
