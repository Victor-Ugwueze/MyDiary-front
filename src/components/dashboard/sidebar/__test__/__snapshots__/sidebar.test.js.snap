// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the sidebar component should render sidebar of dashboard page 1`] = `
<div
  id="left-section"
>
  <ul
    className="nav-tabs hide-sidebar "
    role="tablist"
  >
    <li>
      <div
        className="nav-item-row"
      >
        <div
          className="icon"
        >
          <img
            className="add-entry"
            data-target="add-new-entry"
            src="/images/icons/diary_add_white.png"
          />
        </div>
        <div
          className="text"
        >
          <h3
            className="add-entry"
            data-target="add-new-entry"
          >
            New Entry
          </h3>
        </div>
      </div>
    </li>
    <li>
      <div
        className="nav-item-row nav-it active dairy"
        data-target="diary"
      >
        <img
          className="icon"
          data-target="diary"
          src="/images/icons/diary_icon_list_white.png"
        />
        <h3
          className="text"
          data-target="diary"
          role="tab"
        >
          My Diary
        </h3>
      </div>
    </li>
    <li>
      <div
        className="nav-item-row undefined nav-it settings"
        data-target="settings"
      >
        <img
          className="icon"
          data-target="settings"
          src="/images/icons/settings_icon.png"
        />
        <h3
          className="text"
          data-target="settings"
          role="tab"
        >
          Settings
        </h3>
      </div>
    </li>
    <li>
      <div
        className="nav-item-row nav-it get-profile undefined"
        data-target="profile"
      >
        <img
          className="icon"
          data-target="profile"
          src="/images/icons/profile_icon_white.png"
        />
        <h3
          className="text"
          data-target="profile"
          role="tab"
        >
          Profile
        </h3>
      </div>
    </li>
    <div
      className="nav-item-row nav-it"
      id="logout"
    >
      <div
        className="icon"
      >
        <img
          src="/images/icons/logout.png"
        />
      </div>
      <div>
        <Link
          replace={false}
          to="/logout"
        >
          Log out
        </Link>
      </div>
    </div>
  </ul>
</div>
`;
